### DISCUSSION TITLE
Does time spent in a queued state count for `execution_timeout`?

### DISCUSSION THREAD

**Discussion by falonso-alo** [apache/airflow#29251] [Q&A]
`execution_timeout`'s [documentation](https://airflow.apache.org/docs/apache-airflow/1.10.6/_api/airflow/operators/index.html#airflow.operators.BaseOperator) states the following:

> max time allowed for the execution of this task instance, if it goes beyond it will raise and fail.

What constitutes "execution"? Is it only the time when the task is in a running state? Or does time in the queued state count against the `execution_timeout`? 

According to [this issue](https://github.com/apache/airflow/discussions/28714#discussioncomment-4685765), It seems like it might not!

> On average, we found 1~3 task instances stuck in queued states per day, and those tasks did not conform execution_timeout or write any task logs, because they had not been launched by any celery worker!

Alternatively, does the `dagrun_timeout` include the time where tasks are in a queued state? Documentation [here](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/models/dag/index.html#airflow.models.dag.DAG).

> specify how long a DagRun should be up before timing out / failing, so that new DagRuns can be created. The timeout is only enforced for scheduled DagRuns.

On my team, we've seen some tasks periodically get stuck in a queued state and I'm hoping that setting a timeout value at the task or DAG level will kill the task, but I'm not sure. Please let me know if there is any community guidance on the topic.

Thank you!

**Comment by potiuk** [comment]
* No. Execution_timeout is exclusively for execution (including task deferral time in deferrable operators, but that's not your case).
* Yes. Dagrun_timout is calculated from the moment DagRun is created (and it will stop all tasks that are queued if timeout passes)

**Reply by falonso-alo** [reply]
Okay, that makes total sense! If we need to handle tasks getting stuck in queues, we'll have to use the DAG run timeout. 

Thank you so much!
